### ISSUE TITLE
geth 1.9.9 not syncing

### ISSUE THREAD

**Issue by ankita-p17** [ethereum/go-ethereum#20496]
Hi,

After I reported the issue https://github.com/ethereum/go-ethereum/issues/20435 and muir glacier upgrade I updated my geth to 1.9.9 and using the same dataset collected using 1.9.8

#### System information

Geth version: 1.9.9
OS & Version: Linux

#### Expected behaviour
It should start to sync

#### Actual behaviour

```
Jan 03 07:22:39 dcx-geth geth[11833]: INFO [01-03|07:22:39.040] Initializing fast sync bloom             items=11527142 errorrate=0.000 elapsed=48.007s
Jan 03 07:22:31 dcx-geth geth[11833]: INFO [01-03|07:22:31.040] Initializing fast sync bloom             items=9318579 errorrate=0.000 elapsed=40.006s
Jan 03 07:22:23 dcx-geth geth[11833]: INFO [01-03|07:22:23.039] Initializing fast sync bloom             items=7778639 errorrate=0.000 elapsed=32.006s
Jan 03 07:22:15 dcx-geth geth[11833]: INFO [01-03|07:22:15.034] Initializing fast sync bloom             items=7763381 errorrate=0.000 elapsed=24.000s
Jan 03 07:22:07 dcx-geth geth[11833]: INFO [01-03|07:22:07.033] Initializing fast sync bloom             items=4985442 errorrate=0.000 elapsed=16.000s
Jan 03 07:21:59 dcx-geth geth[11833]: INFO [01-03|07:21:59.033] Initializing fast sync bloom             items=1465133 errorrate=0.000 elapsed=8.000s
```

#### Steps to reproduce the behaviour
install geth 1.9.9 and start to sync

Please suggest. I need my geth to sync completely as soon as possible
@fjl @ligi @karalabe

**Comment by ankita-p17** [comment]
Hello, any update on this?

**Comment by karalabe** [comment]
We've been looking into it. One question however, does your node have UDP connectivity? From the logs it seems as if you're completely firewalled off

**Comment by ankita-p17** [comment]
> We've been looking into it. One question however, does your node have UDP connectivity? From the logs it seems as if you're completely firewalled off

No it doesn't have

**Comment by ankita-p17** [comment]
> We've been looking into it. One question however, does your node have UDP connectivity? From the logs it seems as if you're completely firewalled off
But before also it was working fine with these network configurations

**Comment by karalabe** [comment]
We have a fallback in place so if discovery doesn't work for whatever reason, the node still tried to latch itself to the bootnodes. A refactor of the discovery (we're prepping for DNS discovery) broke this fallback mechanism in 1.9.9, so UDP was a must. We've fixed it (and released) in 1.9.10 just now. Please check again.

That said, this is really a very unhealthy fallback, since the bootnodes may or may not have enough slots open. You should really sort out UDP connectivity (at least make sure DNS will work as that will be shipped soon and may solve your problems too).
